service/history/archival_queue_task_executor.go

```diff
@@ -154,52 +156,47 @@ func (e *archivalQueueTaskExecutor) getArchiveTaskRequest(
 		return nil, err
 	}
 
+	var historyURI, visibilityURI carchiver.URI
 	var targets []archival.Target
 	if e.shardContext.GetArchivalMetadata().GetVisibilityConfig().ClusterConfiguredForArchival() &&
 		namespaceEntry.VisibilityArchivalState().State == enumspb.ARCHIVAL_STATE_ENABLED {
 		targets = append(targets, archival.TargetVisibility)
+		visibilityURIString := namespaceEntry.VisibilityArchivalState().URI
+		visibilityURI, err = carchiver.NewURI(visibilityURIString)
+		if err != nil {
+			e.metricsHandler.Counter(metrics.ArchivalTaskInvalidURI.GetMetricName()).Record(
+				1,
+				metrics.NamespaceTag(namespaceName.String()),
+				metrics.FailureTag("invalid_visibility_uri"),
```

Comment:
nit: put it in the common metrics definition?

Reply:
Thanks, done